serve s3: ListObjectsV2 should return CommonPrefixes (directories) with a trailing /

#### What is the problem you are having with rclone?
I am using rclone serve s3 as an encrypting proxy in front of a Ceph/RGW S3 storage. According to the AWS docs 
the ListObjectsV2 API call shoud return CommonPrefixes (directories) with a trailing forward slash. 
But the s3 implementation of rclone serve s3 does not add trailing slashes to directories returned as CommonPrefixes in the ListObjectsV2 call. 

Reference: [API_ListObjectsV2_ResponseSyntax](https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html#API_ListObjectsV2_ResponseSyntax)

Implications: some tools (e.g. pgbarman) rely on that to find backups stored in s3 and e.g. filter out all returned prefixes not containing a trailing /, assuming these are not directories.

Reference: [pgbarman-code](https://github.com/EnterpriseDB/barman/blob/d68a70eb6d5473fe29bc6895dee94a723ab128cb/barman/cloud.py#L2130)

#### What is your rclone version (output from `rclone version`)
1.69.1


#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
Linux amd64


#### Which cloud storage system are you using? (e.g. Google Drive)
Ceph S3

Setup: client (mostly backup tools like pgbarman, velero ...) --> rclone serve s3 crypt_backend: --> Ceph/RGW S3

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)
HTTP Request from pgbarman:
```
2025-03-20 09:19:46,920 [11] DEBUG: Sending http request: <AWSPreparedRequest stream_output=False, method=GET, url=https://host.docker.internal:4443/cnpg-backups?list-type=2&prefix=cnpg-cluster-zitadel%2Fbase%2F&delimiter=%2F&encoding-type=url, headers={'User-Agent': b'Boto3/1.34.46 md/Botocore#1.34.46 ua/2.0 os/linux#6.10.14-linuxkit md/arch#aarch64 lang/python#3.12.3 md/pyimpl#CPython cfg/retry-mode#legacy Botocore/1.34.46 Resource', 'X-Amz-Date': b'20250320T091946Z', 'X-Amz-Content-SHA256': b'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855', 'Authorization': b'AWS4-HMAC-SHA256 Credential=****/20250320/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=***', 'amz-sdk-invocation-id': b'd82e2ad5-4bb6-4695-9337-050bd405aede', 'amz-sdk-request': b'attempt=1'}>
``` 


#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)
Response logged from pgbarman:
```
2025-03-20 09:19:47,185 botocore.parsers [DEBUG] Response body:
b'<?xml version="1.0" encoding="UTF-8"?>\n<ListBucketResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">\n <Name>cnpg-backups</Name>\n <IsTruncated>false</IsTruncated>\n <Delimiter>/</Delimiter>\n <Prefix>cnpg-cluster-zitadel/base/</Prefix>\n <MaxKeys>1000</MaxKeys>\n <CommonPrefixes>\n <Prefix>cnpg-cluster-zitadel/base/20250319T152145</Prefix>\n </CommonPrefixes>\n <CommonPrefixes>\n <Prefix>cnpg-cluster-zitadel/base/20250319T154154</Prefix>\n </CommonPrefixes>\n <CommonPrefixes>\n <Prefix>cnpg-cluster-zitadel/base/20250319T155715</Prefix>\n </CommonPrefixes>\n <EncodingType>url</EncodingType>\n <KeyCount>3</KeyCount>\n</ListBucketResult>'
...
025-03-20 09:19:47,188 [11] DEBUG: Event after-call.s3.ListObjectsV2: calling handler <function decode_list_object_v2 at 0xffffb875cb80>
2025-03-20 09:19:47,188 botocore.hooks [DEBUG] Event after-call.s3.ListObjectsV2: calling handler <function decode_list_object_v2 at 0xffffb875cb80>
2025-03-20 09:19:47,188 [11] ERROR: Barman cloud backup show exception: Unknown backup 'backup-20250319155714' for server 'cnpg-cluster-zitadel'
2025-03-20 09:19:47,188 root [ERROR] Barman cloud backup show exception: Unknown backup 'backup-20250319155714' for server 'cnpg-cluster-zitadel'

```
As you can see the directory names are returned without a trailing / and thus filtered out by pgbarman and in consequence no backups are found and an error is returned. 

#### Suggested fix
I think this should be fixed in the rclone s3 implementation rather than in the client making this (maybe not so smart assumption) because AWS/Ceph S3 (and I guess all others) behave this way. 
I already fixed and tested this locally:

```diff
diff --git a/cmd/serve/s3/list.go b/cmd/serve/s3/list.go
index 9b0a2e11e..93000907a 100644
--- a/cmd/serve/s3/list.go
+++ b/cmd/serve/s3/list.go
@@ -28,7 +28,8 @@ func (b *s3Backend) entryListR(_vfs *vfs.VFS, bucket, fdPath, name string, addPr

                if entry.IsDir() {
                        if addPrefix {
-                               response.AddPrefix(objectPath)
+                               prefixWithTrailingSlash := objectPath + "/"
+                               response.AddPrefix(prefixWithTrailingSlash)
                                continue
                        }
                        err := b.entryListR(_vfs, bucket, path.Join(fdPath, object), "", false, response)
```
but I am not really confident that this fix won't break other things. So I filed an issue rather than a PR to discuss this with somebody that is more in the serve s3 implementation. 

<!--- Please keep the note below for others who read your bug report. -->

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

--- a/cmd/serve/s3/list.go
+++ b/cmd/serve/s3/list.go
@@ -28,7 +28,8 @@ func (b *s3Backend) entryListR(_vfs *vfs.VFS, bucket, fdPath, name string, addPr
 
 		if entry.IsDir() {
 			if addPrefix {
-				response.AddPrefix(objectPath)
+				prefixWithTrailingSlash := objectPath + "/"
+				response.AddPrefix(prefixWithTrailingSlash)
 				continue
 			}
 			err := b.entryListR(_vfs, bucket, path.Join(fdPath, object), "", false, response)

